arbnode/mel/extraction/message_extraction_function_test.go

```diff
@@ -0,0 +1,375 @@
+		},
+		{
+			name:               "parse sequencer message fails",
+			melStateParentHash: prevParentBlockHash,
+			lookupBatches:      successfulLookupBatches,
+			lookupDelayedMsgs:  successfulLookupDelayedMsgs,
+			serializer:         successfulSerializer,
+			parseReport:        successfulParseReport,
+			parseSequencerMsg:  failingParseSequencerMsg,
+			expectedError:      "failed to parse sequencer message",
+		},
+		{
+			name:                 "extracting batch messages fails",
+			melStateParentHash:   prevParentBlockHash,
+			lookupBatches:        successfulLookupBatches,
+			lookupDelayedMsgs:    successfulLookupDelayedMsgs,
+			serializer:           successfulSerializer,
+			parseReport:          successfulParseReport,
+			parseSequencerMsg:    successfulParseSequencerMsg,
+			extractBatchMessages: failingExtractBatchMessages,
+			expectedError:        "failed to extract batch messages",
+		},
+		{
+			name:                 "OK",
+			melStateParentHash:   prevParentBlockHash,
+			lookupBatches:        successfulLookupBatches,
+			lookupDelayedMsgs:    successfulLookupDelayedMsgs,
+			serializer:           successfulSerializer,
+			parseReport:          successfulParseReport,
+			parseSequencerMsg:    successfulParseSequencerMsg,
+			extractBatchMessages: successfulExtractBatchMessages,
+			expectedMsgCount:     2,
+			expectedDelayedSeen:  1,
+			expectedMessages:     2,
+			expectedDelayedMsgs:  1,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			header := createBlockHeader(prevParentBlockHash)
+			melState := createMelState(tt.melStateParentHash)
+			txsFetcher := &mockTxsFetcher{}
+
+			var postState *mel.State
+			var messages []*arbostypes.MessageWithMetadata
+			var delayedMessages []*mel.DelayedInboxMessage
+			var err error
+
+			if tt.useExtractMessages {
+				// Test the public ExtractMessages function
+				postState, messages, delayedMessages, err = ExtractMessages(
+					ctx,
+					melState,
+					header,
+					nil,
+					nil,
+					nil,
+					txsFetcher,
+				)
+			} else {
+				// Test the internal extractMessagesImpl function
+				postState, messages, delayedMessages, err = extractMessagesImpl(
+					ctx,
+					melState,
+					header,
+					nil,
+					nil,
+					txsFetcher,
+					nil,
+					nil,
+					tt.lookupBatches,
+					tt.lookupDelayedMsgs,
+					tt.serializer,
+					tt.extractBatchMessages,
+					tt.parseSequencerMsg,
+					tt.parseReport,
+				)
+			}
+
+			if tt.expectedError != "" {
+				require.ErrorContains(t, err, tt.expectedError)
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, tt.expectedMsgCount, postState.MsgCount)
+			require.Equal(t, tt.expectedDelayedSeen, postState.DelayedMessagedSeen)
+			require.Len(t, messages, tt.expectedMessages)
+			require.Len(t, delayedMessages, tt.expectedDelayedMsgs)
+		})
+	}
+}
+
+var (
+	// Common setup function
+	createMelState = func(parentHash common.Hash) *mel.State {
```

Comment:
Why use `var (` instead of just defining package private functions?

```
func createMelState(parentHash common.Hash) *mel.State {
	melState := &mel.State{
		ParentChainBlockHash: parentHash,
	}
	return melState
}
```